### ISSUE TITLE
can not unmarshal custom(org.apache.cassandra.db.marshal.VectorType(org.apache.cassandra.db.marshal.FloatType, 1536))

### ISSUE THREAD

**Issue by 7flash** [datastax/gocql-astra#17]
I have following script:

```
package main

import (
	"flag"
	"fmt"
	"os"
	"time"
	"log"

	gocqlastra "github.com/datastax/gocql-astra"
	"github.com/gocql/gocql"
)

func main() {
	// Parse the user ID from the command-line argument
	flag.Parse()
	if len(flag.Args()) != 1 {
		fmt.Printf("Usage: %s <user-id>\n", os.Args[0])
		os.Exit(1)
	}
	userID := flag.Arg(0)

	var cluster *gocql.ClusterConfig
	cluster, err := gocqlastra.NewClusterFromBundle(os.Getenv("ASTRA_DB_SECURE_BUNDLE_PATH"),
		"token", os.Getenv("ASTRA_DB_APPLICATION_TOKEN"), 30*time.Second)
	
	if err != nil {
		panic("unable to load the bundle")
	}
	cluster.Timeout = 30 * time.Second
	
	session, err := gocql.NewSession(*cluster)
	if err != nil || session == nil {
		log.Fatalf("unable to connect to the session: %v", err)
	}
	defer session.Close()
	fmt.Println("Connected to the cluster.")

	// Fetch the vector embedding for the given user ID
	var vectorEmbedding []byte
	if err := session.Query("SELECT query_vector_value FROM keyspace_first.profiles WHERE key=(1, ?)", userID).Scan(&vectorEmbedding); err != nil {
		fmt.Printf("Failed to fetch vector embedding for user ID '%s': %v\n", userID, err)
		os.Exit(1)
	}
}
```

Unfortunately it fails to deserialize vector embedding, tried with []float64 instead of []byte and still same error.

```
go run cli.go fbfba2cc-9d6c-11ee-9851-0242ac1c0009
```

```
Connected to the cluster.
Failed to fetch vector embedding for user ID 'fbfba2cc-9d6c-11ee-9851-0242ac1c0009': can not unmarshal custom(org.apache.cassandra.db.marshal.VectorType(org.apache.cassandra.db.marshal.FloatType, 1536)) into *[]uint8
exit status 1
```

**Comment by 7flash** [comment]
It works in python though well:

```
from cassandra.cluster import Cluster
from cassandra.auth import PlainTextAuthProvider
import json

cloud_config= {
  'secure_connect_bundle': ''
}

with open(") as f:
	secrets = json.load(f)

CLIENT_ID = secrets["clientId"]
CLIENT_SECRET = secrets["secret"]

auth_provider = PlainTextAuthProvider(CLIENT_ID, CLIENT_SECRET)
cluster = Cluster(cloud=cloud_config, auth_provider=auth_provider)
session = cluster.connect()

row = session.execute("SELECT query_vector_value FROM keyspace_first.profiles WHERE key=(1, 'fbfba2cc-9d6c-11ee-9851-0242ac1c0009')").one()
if row:
  print(row[0])
else:
  print("An error occurred.")
```

**Comment by 7flash** [comment]
Based on this line I would expect using []float32 to be a correct solution but it still has same error

https://github.com/gocql/gocql/blob/34fdeebefcbf183ed7f916f931aa0586fdaa1b40/helpers.go#L37

```
Failed to fetch vector embedding for user ID 'fbfba2cc-9d6c-11ee-9851-0242ac1c0009': can not unmarshal custom(org.apache.cassandra.db.marshal.VectorType(org.apache.cassandra.db.marshal.FloatType, 1536)) into *[]float32
```

**Comment by absurdfarce** [comment]
Hey @7flash , thanks for the report!

Adding support for the vector type isn't something that falls within the functionality gocql-astra offers.  In order to support this type you would need to add support for vectors directly to [gocql](https://github.com/gocql/gocql) itself.  I would encourage you to [file an issue](https://github.com/gocql/gocql/issues) with the gocql project asking for support for this type.  While vector support has been introduced in DataStax Astra it is also [coming to Apache Cassandra as well](https://issues.apache.org/jira/browse/CASSANDRA-18557) so it's likely the gocql project will be interested.

Note that support for the vector type was added to the Python driver in [PYTHON-1352](https://datastax-oss.atlassian.net/browse/PYTHON-1352) and released in [version 3.28.0](https://github.com/datastax/python-driver/blob/master/CHANGELOG.rst#3280).  That's why you're able to work with this data type using the Python driver.

Thanks again!

**Comment by absurdfarce** [comment]
Thanks for filing the feature request with gocql @7flash!  I'm going to close this ticket out to avoid any confusion; any work to address this issue will be managed under the gocql ticket.
